### ISSUE TITLE
Loading data file with no data results in an error for the BalancingLearner 

### ISSUE THREAD

**Issue by basnijholt** [python-adaptive/adaptive#148]
This happens when one has many learners and some learners have been saved even though there is no data.

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-6-224a1822286c> in <module>
     23 combos = {'z_y': z_ys, 'ratio': [4, 8, 16]}
     24 learner = adaptive.BalancingLearner.from_product(gap_wrapper, Learner, kwargs, combos)
---> 25 learner.load(fnames)

/opt/conda/lib/python3.6/site-packages/adaptive/learner/balancing_learner.py in load(self, fname, compress)
    379         else:
    380             for l in self.learners:
--> 381                 l.load(fname(l), compress=compress)
    382 
    383     def _get_data(self):

/opt/conda/lib/python3.6/site-packages/adaptive/learner/data_saver.py in load(self, fname, compress)
     64         # We copy this method because the 'DataSaver' is not a
     65         # subclass of the 'BaseLearner'.
---> 66         BaseLearner.load(self, fname, compress)
     67 
     68 

/opt/conda/lib/python3.6/site-packages/adaptive/learner/base_learner.py in load(self, fname, compress)
    136         with suppress(FileNotFoundError, EOFError):
    137             data = load(fname, compress)
--> 138             self._set_data(data)
    139 
    140     def __getstate__(self):

/opt/conda/lib/python3.6/site-packages/adaptive/learner/data_saver.py in _set_data(self, data)
     52     def _set_data(self, data):
     53         learner_data, self.extra_data = data
---> 54         self.learner._set_data(learner_data)
     55 
     56     @copy_docstring_from(BaseLearner.save)

/opt/conda/lib/python3.6/site-packages/adaptive/learner/learner1D.py in _set_data(self, data)
    655 
    656     def _set_data(self, data):
--> 657         self.tell_many(*zip(*data.items()))
    658 
    659 

TypeError: tell_many() missing 2 required positional arguments: 'xs' and 'ys'
```

**Comment by basnijholt** [comment]
The solution would be to add a `TypeError` to the suppress. 

Of course https://github.com/python-adaptive/adaptive/issues/15 will remain a problem.

**Comment by basnijholt** [comment]
Closed by #149.
